src/goto-symex/goto_symex_state.cpp

```diff
@@ -217,16 +215,19 @@ void goto_symex_statet::assignment(
   // the type might need renaming
   rename(lhs.type(), l1_identifier, ns);
   lhs.update_type();
-  const validation_modet vm = global_validation_mode;
-  DATA_CHECK(!check_renaming_l1(lhs), "lhs renaming failed on l1");
-
+  const validation_modet local_validation_mode = validation_modet::INVARIANT;
+  if(run_validation_checks)
+  {
+    const validation_modet vm = local_validation_mode;
+    DATA_CHECK(!check_renaming_l1(lhs), "lhs renaming failed on l1");
```

Comment:
Hopefully that pain will go away soon when `DATA_CHECK` takes three arguments...